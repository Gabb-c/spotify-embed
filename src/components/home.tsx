import React from 'react';

export const Home: React.FC = (): JSX.Element => (
  <>
    <iframe
      src="https://open.spotify.com/embed/album/1DFixLWuPkv3KT3TnV35m3"
      width="300"
      height="380"
      frameBorder="0"
      allowTransparency
      aria-label="spotify-embed"
      allow="encrypted-media"
    ></iframe>
  </>
);
