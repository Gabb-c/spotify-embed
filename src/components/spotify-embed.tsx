import { Default, Compact, urlParser, uriParser } from '../utils';

export interface SpotifyEmbedProps
  extends Omit<
    React.IframeHTMLAttributes<HTMLIFrameElement>,
    'allow-transparency' | 'frameBorder' | 'aria-label' | 'allow'
  > {
  /**
   * The size of the embed
   * @defaultValue 'default'
   */
  size?: 'default' | 'compact' | 'custom';
  /**
   * The Spotify resource URL/URI
   * ( track | playlist | album | artist | show | episode )
   */
  src?: string;
  /**
   * The embed theme
   * @defaultValue 'light'
   */
  theme?: 'light' | 'dark';
}

/**
 * ## Spotfy Embed
 * @param props - Spotify embed properties
 * @returns The Spotify embed
 * @see https://developer.spotify.com/documentation/widgets/
 */
export const SpotifyEmbed: React.FC<SpotifyEmbedProps> = (props): JSX.Element => {
  const {
    size = 'default',
    src = '',
    width = Default.WIDTH,
    height = Default.HEIGHT,
    theme = 'light',
  } = props;

  return (
    <>
      <iframe
        {...props}
        src={urlParser(src) === '' ? uriParser(src, theme) : urlParser(src, theme)}
        width={size === 'default' ? Default.WIDTH : size === 'custom' ? width : Compact.WIDTH}
        height={size === 'default' ? Default.HEIGHT : size === 'custom' ? height : Compact.HEIGHT}
        frameBorder="0"
        aria-label="spotify-embed"
        allow="autoplay; transparency; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
      />
    </>
  );
};

// storybook default props
SpotifyEmbed.defaultProps = {
  size: 'default',
  src: '',
  width: Default.WIDTH,
  height: Default.HEIGHT,
  theme: 'light',
};
