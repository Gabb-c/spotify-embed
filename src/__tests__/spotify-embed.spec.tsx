import { SpotifyEmbed } from "../components";
import { render, screen } from "@testing-library/react";
import { cleanup } from "@testing-library/react";
import React from "react";
import { afterEach, describe, expect, it } from "vitest";

describe("Spotify Embed", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render and be in the document (URL)", () => {
    render(<SpotifyEmbed src="https://open.spotify.com/track/5mslcDXYQbmOFL6s031NvN" />);

    const embed = screen.queryByLabelText("spotify-embed");

    expect(embed).toBeDefined();
  });

  it("should match snapshot (URL)", () => {
    render(<SpotifyEmbed src="https://open.spotify.com/track/5mslcDXYQbmOFL6s031NvN" />);

    const embed = screen.queryByLabelText("spotify-embed");

    expect(embed).toMatchSnapshot();
  });

  it("should render and be in the document (URI)", () => {
    render(<SpotifyEmbed src="spotify:track:5mslcDXYQbmOFL6s031NvN" />);

    const embed = screen.queryByLabelText("spotify-embed");

    expect(embed).toBeDefined();
  });

  it("should match snapshot (URI)", () => {
    render(<SpotifyEmbed src="spotify:track:5mslcDXYQbmOFL6s031NvN" />);

    const embed = screen.queryByLabelText("spotify-embed");

    expect(embed).toMatchSnapshot();
  });
});
