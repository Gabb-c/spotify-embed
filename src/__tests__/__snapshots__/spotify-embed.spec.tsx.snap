// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spotify Embed > should match snapshot (URI) 1`] = `
<iframe
  allow="autoplay; transparency; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
  aria-label="spotify-embed"
  height="380"
  src="https://open.spotify.com/embed/track/5mslcDXYQbmOFL6s031NvN?theme=1"
  theme="light"
  width="300"
/>
`;

exports[`Spotify Embed > should match snapshot (URL) 1`] = `
<iframe
  allow="autoplay; transparency; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
  aria-label="spotify-embed"
  height="380"
  src="https://open.spotify.com/embed/track/5mslcDXYQbmOFL6s031NvN?theme=1"
  theme="light"
  width="300"
/>
`;
